dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleStatsQuery.java

```diff
@@ -42,44 +42,49 @@ public abstract class OracleStatsQuery {
   @Nonnull
   abstract String name();
 
-  @Nonnull
-  abstract String queryText();
-
   @Nonnull
   abstract StatsSource statsSource();
 
   @Nonnull
-  static OracleStatsQuery createAwr(String name, Duration queriedDuration) throws IOException {
+  static OracleStatsQuery createAwr(String name, Duration queriedDuration) {
     return create(false, queriedDuration, name, AWR);
   }
 
   @Nonnull
-  static OracleStatsQuery createNative(String name, boolean isRequired, Duration queriedDuration)
-      throws IOException {
+  static OracleStatsQuery createNative(String name, boolean isRequired, Duration queriedDuration) {
     return create(isRequired, queriedDuration, name, NATIVE);
   }
 
   @Nonnull
-  static OracleStatsQuery createStatspack(String name, Duration queriedDuration)
-      throws IOException {
+  static OracleStatsQuery createStatspack(String name, Duration queriedDuration) {
     return create(false, queriedDuration, name, STATSPACK);
   }
 
   private static OracleStatsQuery create(
-      boolean isRequired, Duration queriedDuration, String name, StatsSource statsSource)
-      throws IOException {
-    String path = String.format("oracle-stats/cdb/%s/%s.sql", statsSource.value, name);
-    return new AutoValue_OracleStatsQuery(
-        isRequired, queriedDuration, name, loadFile(path), statsSource);
+      boolean isRequired, Duration queriedDuration, String name, StatsSource statsSource) {
+    return new AutoValue_OracleStatsQuery(isRequired, queriedDuration, name, statsSource);
+  }
+
+  @Nonnull
+  String queryText() {
+    String source = statsSource().value;
+    String tenantSetup = "cdb";
+    String path = String.format("oracle-stats/%s/%s/%s.sql", tenantSetup, source, name());
+    return loadFile(path);
```

Comment:
I understand the change with catching `IOException` inside the `loadFile`, but it's unclear to me, why you are moving the loading from `create` to the `queryText` method. I think the previous implementation was better in this aspect. I might be missing something here. I would recommend keeping the getter simple and restructuring the code in such a way that the SQL is read just once, no matter how many times the getter is invoked. (even if in practice it is currently invoked just once)

Reply:
Done, I restored the property. The `create` method throwing an exception was inconvenient to use, partly because the exception was checked and partly because it needed to reference a valid resource file whenever it's called. The first point is gone, if the second ever starts causing problems I'll take a second look at this. In that case, maybe Memoized can be used to avoid re-reading the SQL file.